frontend/src/components/explorer/mapViewer/ThreeDviewer.vue

```diff
@@ -52,30 +91,137 @@ export default {
+      this.controller.setData({
+        graphData: customizedNetwork || this.network,
+        nodeTextures: [
+          { group: 'e', sprite: '/sprite_round.png' },
           { group: 'r', sprite: '/sprite_square.png' },
-          { group: 'm', sprite: '/sprite_triangle.png' }],
-        15);
+          { group: 'm', sprite: '/sprite_triangle.png' },
+        ],
+        nodeSize: 15,
+      });
+      this.controller.setNodeSelectCallback(this.selectElement);
+      this.controller.setBackgroundColor(this.backgroundColor);
+      this.controller.setUpdateCameraCallback(this.updateURLCoords);
+      this.processURLQuery();
+    },
+    processURLQuery() {
+      this.$store.dispatch('maps/initFromQueryParams', this.$route.query);
+      const { lx, ly, lz } = this.coords;
+      this.controller.setCamera({ x: lx, y: ly, z: lz });
+
+      const id = this.queryParams.sel;
+
+      if (id) {
+        setTimeout(() => {
+          this.controller.selectBy({ id });
+        }, 150);
+      }
+    },
+    async selectElement(element) {
+      const [id, type] = this.getElementIdAndType(element);
+      const selectionData = { type, data: null, error: false };
+
+      this.$emit('startSelection');
+      try {
+        const payload = {
+          model: this.model.apiName,
+          version: this.model.apiVersion,
+          type,
+          id,
+        };
+        await this.$store.dispatch('maps/getSelectedElement', payload);
+        const data = this.selectedElement;
+        selectionData.data = data;
+        this.$emit('updatePanelSelectionData', selectionData);
+        this.$emit('endSelection', true);
+      } catch {
+        this.$emit('updatePanelSelectionData', selectionData);
+        this.$set(selectionData, 'error', true);
+        this.$emit('endSelection', false);
+      }
+    },
+    applyColorsAndRenderNetwork(levels) {
+      const nodes = this.network.nodes.map((node) => {
+        let color = colorToRGBArray('#9df');
+
+        if (node.g === 'e') {
+          if (Object.keys(levels).length === 0) {
+            color = colorToRGBArray('#feb');
+          } else {
+            const partialID = node.id.split('-')[0];
+            const key = levels[partialID] !== undefined ? partialID : 'n/a';
+            color = colorToRGBArray(levels[key][0]);
+          }
+        }
+
+        if (node.g === 'r') {
+          color = colorToRGBArray('#fff');
+        }
+
+        return {
+          ...node,
+          color,
+        };
+      });
+
+      this.renderNetwork({
+        nodes,
+        links: this.network.links,
+      });
+    },
+    updateURLCoords({ x, y, z }) {
+      const payload = {
+        ...this.coords,
+        lx: x,
+        ly: y,
+        lz: z,
+      };
+      this.$store.dispatch('maps/setCoords', payload);
     },
     resetNetwork() {
       const viewer = document.getElementById('viewer3d');
       viewer.innerHTML = '';
     },
+    zoomIn() {
+      console.log('zoom in');
```

Comment:
Console logs break the production build, but I guess this meant to be fixed immediately?

Reply:
Yes, this is bad practice but considering that there are so many bugs left for the map viewer I think we can ignore it for now.

Reply:
Alright, I've checked and as you say this issue is already noted under MA-897.